polaris/polaris/utils.py

```diff
@@ -420,18 +445,68 @@ def create_transaction_envelope(transaction, source_account) -> TransactionEnvel
         source_account=source_account,
         network_passphrase=settings.STELLAR_NETWORK_PASSPHRASE,
         base_fee=base_fee,
-    ).append_payment_op(
-        destination=transaction.stellar_account,
-        asset_code=transaction.asset.code,
-        asset_issuer=transaction.asset.issuer,
-        amount=str(payment_amount),
-        source=transaction.asset.distribution_account,
     )
+    if (
+        transaction.claimable_balance_supported
+        and transaction.status == Transaction.STATUS.pending_trust
+    ):
```

Comment:
Instead of checking `status`, we'll need to check if the destination account has a trust-line.